cabal-install/Distribution/Solver/Modular/Log.hs

```diff
@@ -72,17 +77,21 @@ logToProgress mbj l = let
     -- exhaustiveness and first conflict set.
     go :: Progress Message a b
        -> Progress Message a b
-       -> Progress String (Bool, Maybe (ConflictSet QPN)) b
+       -> Progress String (Exhaustive, ConflictSet QPN, ConflictMap) b
        -> Progress String String b
     go ms (Step _ ns) (Step x xs)           = Step x (go ms ns xs)
     go ms r           (Step x xs)           = Step x (go ms r  xs)
-    go ms _           (Fail (exh, Just cs)) = Fail $
-                                              "Could not resolve dependencies:\n" ++
-                                              unlines (messages $ showMessages (L.foldr (\ v _ -> v `CS.member` cs) True) False ms) ++
-                                              (if exh then "Dependency tree exhaustively searched.\n"
-                                                      else "Backjump limit reached (" ++ currlimit mbj ++
-                                                               "change with --max-backjumps or try to run with --reorder-goals).\n")
-                                                  where currlimit (Just n) = "currently " ++ show n ++ ", "
-                                                        currlimit Nothing  = ""
+    go ms _           (Fail (exh, cs, cm))  = Fail $
+        "Could not resolve dependencies:\n" ++
+        unlines (messages $ showMessages (L.foldr (\ v _ -> v `CS.member` cs) True) False ms) ++
```

Comment:
It looks like `cs` is the first conflict set when the backjump limit is reached and the last conflict set when the search is exhaustive.  Shouldn't this line always filter the short log using the first conflict set?  The final conflict set might not contain the first error, which would cause the error to be filtered out.

Another option is to remove the filtered log altogether, to avoid showing two possibly unrelated errors.